--- opal-3.8.1/plugins/video/common/dyna.h.orig	2010-05-07 08:01:09.000000000 +0300
+++ opal-3.8.1/plugins/video/common/dyna.h	2010-11-11 13:11:19.387936827 +0200
@@ -41,6 +41,8 @@
 #ifndef __DYNA_H__
 #define __DYNA_H__ 1
 
+#define __STDC_CONSTANT_MACROS
+
 #include "ffmpeg.h"
 	  
 /////////////////////////////////////////////////////////////////
--- opal-3.8.1/plugins/video/H.263-1998/h263-1998.h.orig	2010-05-07 08:01:09.000000000 +0300
+++ opal-3.8.1/plugins/video/H.263-1998/h263-1998.h	2010-11-11 13:13:41.917936830 +0200
@@ -49,6 +49,8 @@
 #ifndef __H263P_1998_H__
 #define __H263P_1998_H__ 1
 
+#define __STDC_CONSTANT_MACROS
+
 #include <codec/opalplugin.h>
 #include "h263pframe.h"
 #include "rfc2190.h"
--- opal-3.8.1/plugins/video/H.263-1998/h263pframe.h.orig	2010-05-07 08:01:09.000000000 +0300
+++ opal-3.8.1/plugins/video/H.263-1998/h263pframe.h	2010-11-11 13:15:24.184603496 +0200
@@ -23,6 +23,8 @@
 #ifndef __H263PFrame_H__
 #define __H263PFrame_H__ 1
 
+#define __STDC_CONSTANT_MACROS
+
 #include <vector>
 #include <string.h>
 #include "rtpframe.h"
diff -ur opal-3.6.8.orig//plugins/video/H.264/h264-x264.h opal-3.6.8/plugins/video/H.264/h264-x264.h
--- opal-3.6.8.orig//plugins/video/H.264/h264-x264.h	2010-05-27 08:52:46.000000000 +0200
+++ opal-3.6.8/plugins/video/H.264/h264-x264.h	2010-08-25 12:47:19.218239108 +0200
@@ -39,6 +39,8 @@
 #ifndef __H264_X264_H__
 #define __H264_X264_H__ 1
 
+#define __STDC_CONSTANT_MACROS
+
 #include "plugin-config.h"
 
 #include <stdarg.h>
