--- ./opal_inc.mak.org	Wed Sep 11 07:55:40 2002
+++ ./opal_inc.mak	Thu Apr 24 13:52:42 2003
@@ -54,15 +54,15 @@
 
 
 LIBDIRS += $(OPALDIR)
-
+PWLIB_MAKE=$(PWLIBDIR)/make
 
 ifdef LIBRARY_MAKEFILE
-include $(PWLIBDIR)/make/unix.mak
+include $(PWLIB_MAKE)/unix.mak
 else
 ifdef NOTRACE
 OBJDIR_SUFFIX := n
 endif
-include $(PWLIBDIR)/make/ptlib.mak
+include $(PWLIB_MAKE)/ptlib.mak
 endif
 
 
--- ./Makefile.org	Tue Apr  8 13:46:35 2003
+++ ./Makefile	Thu Apr 24 13:57:17 2003
@@ -382,7 +382,7 @@
 
 ####################################################
 
-include $(PWLIBDIR)/make/common.mak
+include $(PWLIB_MAKE)/common.mak
 
 ####################################################
 
@@ -396,7 +396,7 @@
 LIB_FILENAME=$(OPAL_FILE)
 
 
-include $(PWLIBDIR)/make/lib.mak
+include $(PWLIB_MAKE)/lib.mak
 
 ####################################################
 
