--- opal-3.4.2/samples/simple/Makefile.orig	2008-11-05 17:42:13.403269358 +0100
+++ opal-3.4.2/samples/simple/Makefile	2008-11-05 17:58:28.336264433 +0100
@@ -53,13 +53,13 @@
 
 # The -I option allows to build the simple application in the same time as opal
 # (as used by distribution packagers)
-CFLAGS += $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config $(PKG_FLAGS) --cflags opal) -I../../include
-LDFLAGS += $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config $(PKG_FLAGS) --libs opal)
+CFLAGS += 
+LDFLAGS += 
 
 
 # This part is we like to be tidy and put all the build products
 # in a sub-directory of a name specific to platform type.
-OBJDIR = $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config $(PKG_FLAGS) --variable=objdir opal)
+OBJDIR = obj
 
 vpath	%.o   $(OBJDIR)
 
--- opal-3.10.8/plugins/LID/IxJ/Makefile.in~	2012-10-05 08:19:44.000000000 +0200
+++ opal-3.10.8/plugins/LID/IxJ/Makefile.in	2012-10-16 22:15:37.582438492 +0200
@@ -31,7 +31,6 @@
 #
 #
 
-STDCCFLAGS =@STDCCFLAGS@
 CXXFLAGS   =@CXXFLAGS@
 
 LID_PLUGIN_DIR=@LID_PLUGIN_DIR@
@@ -65,7 +64,7 @@
 
 $(OBJDIR)/%.o : %.cxx
 	@mkdir -p $(OBJDIR) >/dev/null 2>&1
-	$(Q_CC)$(CXX) -I../../../include $(STDCCFLAGS) $(OPTCCFLAGS) $(CXXFLAGS) -c $< -o $@
+	$(Q_CC)$(CXX) -I../../../include $(OPTCCFLAGS) $(CXXFLAGS) -c $< -o $@
 
 PLUGIN	= ./ixj_lid_pwplugin.so
 
--- opal-3.10.11/plugins/video/common/mpi.cxx.orig	2017-07-12 19:15:10.396689138 +0000
+++ opal-3.10.11/plugins/video/common/mpi.cxx	2017-07-12 19:18:02.036496397 +0000
@@ -132,8 +132,8 @@
   // to the desired one or matches it
   for (i=0; i < MPIs.size(); i++) {
     // we square the value in order to get absolute distances
-    distance = ( abs(MPIs[i].width  - desiredWidth ) *
-                 abs(MPIs[i].height - desiredHeight) );
+    distance = ( abs((int)(MPIs[i].width  - desiredWidth) ) *
+                 abs((int)(MPIs[i].height - desiredHeight) ) );
 
     if (distance < minDistance) {
       minDistance = distance;
