--- opal-3.10.9/plugins/configure.ac.orig	2013-01-24 15:50:58.908808589 +0100
+++ opal-3.10.9/plugins/configure.ac	2013-01-24 16:05:12.328790801 +0100
@@ -517,11 +517,11 @@
         ],
         [
           AC_MSG_RESULT(yes);
-          AC_DEFINE([AV_PICTURE_TYPE_NONE], [(AVPictureType)0])
+          AC_DEFINE([AV_PICTURE_TYPE_NONE], [(AVPictureType)0], [Compatibilty with various versions of the libavcodec library])
         ],
         [
           AC_MSG_RESULT(no)
-          AC_DEFINE([AV_PICTURE_TYPE_NONE], [0])
+          AC_DEFINE([AV_PICTURE_TYPE_NONE], [0], [Compatibilty with various versions of the libavcodec library])
         ]
       )
     ]
--- opal-3.10.9/configure.ac.orig	2012-11-26 11:49:22.000000000 +0100
+++ opal-3.10.9/configure.ac	2013-01-24 16:06:31.778789145 +0100
@@ -400,7 +400,7 @@
 if test "x$OPAL_CAPI" = "xyes" ; then
   AC_CHECK_HEADERS([capi20.h], [OPAL_CAPI=yes], [OPAL_CAPI=no])
   if test "x$OPAL_CAPI" = "xyes" ; then
-    AC_DEFINE([OPAL_CAPI], [1])
+    AC_DEFINE([OPAL_CAPI], [1], [CAPI support])
     OPAL_ADD_CFLAGS_LIBS([], [-lcapi20])
   fi
 fi
