diff -urNad opal-2.1.3/opal_inc.mak.in /tmp/dpep.vc5Q70/opal-2.1.3/opal_inc.mak.in
--- opal-2.1.3/opal_inc.mak.in	2006-01-31 09:10:37.000000000 +0100
+++ /tmp/dpep.vc5Q70/opal-2.1.3/opal_inc.mak.in	2006-02-06 21:22:59.966232274 +0100
@@ -103,11 +103,11 @@
 endif
 
 
-OPAL_BASE  = opal_$(PLATFORM_TYPE)_$(OPAL_SUFFIX)
-OPAL_FILE  = lib$(OPAL_BASE)$(LIB_TYPE).$(LIB_SUFFIX)
+OPAL_BASE  = opal
+OPAL_FILE  = lib$(OPAL_BASE).$(LIB_SUFFIX)
 
 LDFLAGS	    += -L$(OPAL_LIBDIR)
-LDLIBS	    := -l$(OPAL_BASE)$(LIB_TYPE) $(LDLIBS)
+LDLIBS	    := -l$(OPAL_BASE) $(LDLIBS)
 
 STDCCFLAGS  += -I$(OPAL_INCDIR)
 
--- opal-2.2.1/Makefile.in~	2006-02-01 23:45:40.000000000 +0100
+++ opal-2.2.1/Makefile.in	2006-05-18 18:45:26.569385000 +0200
@@ -697,7 +697,7 @@
 ifneq ($(P_SHAREDLIB),1)
   STATIC_LIB_FILE = $(LIBDIR)/$(OPAL_FILE)
 else
-  STATIC_LIB_FILE = $(LIBDIR)/$(subst .$(LIB_SUFFIX),_s.$(STATICLIBEXT),$(OPAL_FILE))
+  STATIC_LIB_FILE = $(LIBDIR)/$(subst .$(LIB_SUFFIX),.$(STATICLIBEXT),$(OPAL_FILE))
 endif
 
 $(LIBDIR)/$(OPAL_FILE) : $(TARGETLIB)
