diff -ur opal-2.2.8/opal_inc.mak.in opal-2.2.8-mak/opal_inc.mak.in
--- opal-2.2.8/opal_inc.mak.in	2007-04-18 19:16:36.345473082 +0200
+++ opal-2.2.8-mak/opal_inc.mak.in	2007-04-18 19:17:46.448447452 +0200
@@ -67,7 +67,9 @@
 HAS_SANGOMA=@HAS_SANGOMA@
 
 
-OPALDIR=@OPALDIR@
+ifndef OPALDIR
+OPALDIR     = /usr
+endif
 PWLIBDIR=@PWLIBDIR@
 
 LIBDIRS += $(OPALDIR)
@@ -106,7 +108,7 @@
 OPAL_BASE  = opal
 OPAL_FILE  = lib$(OPAL_BASE).$(LIB_SUFFIX)
 
-LDFLAGS	    += -L$(OPAL_LIBDIR)
+LDFLAGS	    := -L$(OPAL_LIBDIR) $(LDFLAGS)
 LDLIBS	    := -l$(OPAL_BASE) $(LDLIBS)
 
 STDCCFLAGS  += -I$(OPAL_INCDIR)
@@ -114,6 +116,7 @@
 
 $(TARGET) :	$(OPAL_LIBDIR)/$(OPAL_FILE)
 
+ifdef OPAL_BUILD
 ifndef LIBRARY_MAKEFILE
 
 ifdef DEBUG
@@ -127,6 +130,7 @@
 libs :: $(OPAL_LIBDIR)/$(OPAL_FILE)
 
 endif
+endif
 
 
 # End of file
