--- opal-3.10.11/opal.m4~	2018-09-27 11:03:07.000000000 +0200
+++ opal-3.10.11/opal.m4	2018-09-27 11:04:28.172164490 +0200
@@ -903,7 +903,7 @@ AC_DEFUN([OPAL_FIND_LIBSRTP],
          [
           AC_LANG(C)
           AC_COMPILE_IFELSE([
-                             AC_LANG_PROGRAM([[#include "srtp/srtp.h"]], 
+                             AC_LANG_PROGRAM([[#include "srtp2/srtp.h"]], 
                                              [[
                                                crypto_policy_t p; p.cipher_key_len = SRTP_MASTER_KEY_LEN; return 0; 
                                              ]]
@@ -913,7 +913,7 @@ AC_DEFUN([OPAL_FIND_LIBSRTP],
                             [opal_libsrtp=no])
 
           if test "x${opal_libsrtp}" = "xyes" ; then
-            SRTP_LIBS="-lsrtp"
+            SRTP_LIBS="-lsrtp2"
           fi
            AS_IF([test AS_VAR_GET([opal_libsrtp]) = yes], [$1], [$2])[]
          ])
